Date: Sep 29, 2014 12:50 PM

From: 9cf95fcb3d99a6ba3b9d35fc937eff7cf14b6229 Sep 29, 2014 12:50 PM

Subject: [PATCH] RDKSYSINTSW-1296: Fix building out of tree

        Use pkg-config for pcre dependency instead of -config script.

Source: COMCAST

Upstream-Status: Pending

Signed-off-by: Khem Raj <khem_raj@cable.comcast.com>
---

Index: lighttpd-1.4.35/configure.ac
===================================================================
--- lighttpd-1.4.35.orig/configure.ac	2014-03-06 14:08:00.000000000 +0000
+++ lighttpd-1.4.35/configure.ac	2014-05-13 16:58:30.758471169 +0000
@@ -309,16 +309,14 @@
 AC_MSG_RESULT([$WITH_PCRE])
 
 if test "$WITH_PCRE" != "no"; then
-  AC_PATH_PROG(PCRECONFIG, pcre-config)
-
-  if test x"$PCRECONFIG" != x; then
-    PCRE_LIB=`$PCRECONFIG --libs`
-    CPPFLAGS="$CPPFLAGS `$PCRECONFIG --cflags`"
+  PKG_CHECK_MODULES(PCREPKG, [libpcre], [
+    PCRE_LIB=${PCREPKG_LIBS}
+    CPPFLAGS="$CPPFLAGS ${PCREPKG_CFLAGS}"
     AC_DEFINE([HAVE_LIBPCRE], [1], [libpcre])
     AC_DEFINE([HAVE_PCRE_H], [1], [pcre.h])
-  else
+  ], [
      AC_MSG_ERROR([pcre-config not found, install the pcre-devel package or build with --without-pcre])
-  fi
+  ])
 fi
 
 AC_SUBST(PCRE_LIB)
